nova_vm/src/small_integer.rs

```diff
@@ -0,0 +1,52 @@
+/// 56-bit signed integer.
+#[derive(Clone, Copy)]
+pub struct SmallInteger {
+    data: [u8; 7],
+}
+
+impl std::fmt::Debug for SmallInteger {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", Into::<i64>::into(*self))
+    }
+}
+
+impl SmallInteger {
+    pub const MIN: i64 = -(2 as i64).pow(56) / 2 + 1;
+    pub const MAX: i64 = (2 as i64).pow(56) / 2 - 1;
+
+    pub(crate) fn from_i64_unchecked(value: i64) -> SmallInteger {
+        debug_assert!(value >= Self::MIN && value <= Self::MAX);
+        let mut data: [u8; 7] = [0, 0, 0, 0, 0, 0, 0];
+        let bytes = i64::to_ne_bytes(value);
+        if cfg!(target_endian = "little") {
+            data.copy_from_slice(&bytes[0..7]);
+        } else {
+            data.copy_from_slice(&bytes[1..8]);
+        }
+        Self { data }
+    }
+}
+
+impl TryFrom<i64> for SmallInteger {
+    type Error = ();
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        if value >= Self::MIN && value <= Self::MAX {
+            Ok(Self::from_i64_unchecked(value))
+        } else {
+            Err(())
+        }
+    }
+}
+
+impl Into<i64> for SmallInteger {
+    fn into(self) -> i64 {
+        let Self { data } = self;
+        let mut bytes: [u8; 8] = [0, 0, 0, 0, 0, 0, 0, 0];
+        bytes.copy_from_slice(&data);
```

Comment:
**issue:** This won't work for negative numbers: The top byte needs to have sign extension which this will not.

Also, `copy_from_slice` will assert that the lengths of the slices are the same so I think this should just assert every time (7 vs 8).

Reply:
Yeah my implementation is broken. I had worked on it before you pushed and (wrongly) didn't look into it. Will copy your code.